internal/cliwrapper/cliwrapper.go

```diff
@@ -94,85 +100,86 @@ func (p *cliWrapper) PullModule(fullModuleName string) error {
 	if err != nil {
 		return err
 	}
-	// create module path
+	pipInstallArgs = append(pipInstallArgs, *module)
+
 	modulePath, err := p.GetModulePath(fullModuleName)
 	if err != nil {
 		return err
 	}
-	if err := os.Mkdir(*modulePath, os.ModePerm); err != nil {
-		return err
-	}
 
-	// create venv
-	if _, err := os.Stat(p.pythonFullPath); os.IsNotExist(err) {
-		return fmt.Errorf("python interpreter not found in %s", p.pythonFullPath)
-	}
+	pipPath := filepath.Join(*modulePath, "venv/bin/pip")
+	cmdPip := exec.Command(pipPath, pipInstallArgs...)
 
-	cmdCreateVenv := exec.Command(p.pythonFullPath, "-m", "venv", "venv")
-	cmdCreateVenv.Dir = *modulePath
-	var cmdCreateOut bytes.Buffer
-	cmdCreateVenv.Stderr = &cmdCreateOut
-	if err := cmdCreateVenv.Run(); err != nil {
-		return fmt.Errorf("error while creating venv. Stderr: '%s', err: '%s'", cmdCreateOut.String(), err)
-	} else if cmdCreateOut.Len() > 0 {
-		p.logger.Warningf("Python deployer venv command had stderr output: %s", cmdCreateOut.String())
-	}
+	// Make git non-interactive, so that it never prompts for credentials.
+	// Otherwise, you can hit edge cases where git will wait for manual
+	// authentication causing pip to hang because pip calls `git clone` in
+	// a subprocess.
+	cmdPip.Env = append(os.Environ(), "GIT_TERMINAL_PROMPT=0")
 
-	// pull module
-	pipPath := fmt.Sprintf("%s/venv/bin/pip", *modulePath)
-	cmdPip := exec.Command(pipPath, "install", *module)
-	var cmdPipOut bytes.Buffer
-	cmdPip.Stderr = &cmdPipOut
-	if err := cmdPip.Run(); err != nil {
-		return fmt.Errorf("error while running pip. stderr: '%s', err: '%s'", cmdPipOut.String(), err)
-	} else if cmdPipOut.Len() > 0 {
-		p.logger.Warningf("Python deployer pip command had stderr output: %s", cmdPipOut.String())
+	output, err := cmdPip.Output()
+	if len(output) > 0 {
+		p.logger.Debugf("pip install stdout: %s", output)
+	}
+	if err != nil {
+		return fmt.Errorf("error pip installing %s (%w)", fullModuleName, err)
 	}
 	return nil
```

Comment:
While these lines appear to be attempting to do the same thing, I found a clear degradation in the visibility of the problem. This is a serious usability concern.

Here is me purposefully using a bad hash, then running it with the old and new deployer:

Old (main):
```
joconnel@joconnel-mac engine % go run cmd/arcaflow/main.go -config=$WFPATH/config.yaml -input=$WFPATH/input.yaml -context=$WFPATH
2024-01-09T09:59:41-05:00	debug	source=deployer	pull policy: IfNotPresent
2024-01-09T09:59:41-05:00	debug	source=deployer	pulling module: arcaflow-plugin-wait@git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong
2024-01-09T09:59:44-05:00	error	source=main	Invalid workflow (invalid workflow (failed to load schema for step long_wait (failed to deploy plugin of deployment type 'python' with source 'arcaflow-plugin-wait@git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong' (error while running pip. stderr: '  Running command git clone --filter=blob:none --quiet https://github.com/arcalot/arcaflow-plugin-wait.git /private/var/folders/d_/wv7q0mcs6x9bm10d_jx0j8tr0000gn/T/pip-install-x1ibfwwm/arcaflow-plugin-wait_54e2ce06ded74484b21a454c5dfa13ed
  WARNING: Did not find branch or tag 'afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong', assuming revision or ref.
  Running command git checkout -q afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong
  error: pathspec 'afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong' did not match any file(s) known to git
  error: subprocess-exited-with-error
  
  × git checkout -q afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× git checkout -q afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.3.1 -> 23.3.2
[notice] To update, run: python3.11 -m pip install --upgrade pip
', err: 'exit status 1'))))
exit status 1
```
With changes:
```
joconnel@joconnel-mac engine % go run cmd/arcaflow/main.go -config=$WFPATH/config.yaml -input=$WFPATH/input.yaml -context=$WFPATH
2024-01-09T10:00:23-05:00	debug	source=deployer	pull policy: IfNotPresent
2024-01-09T10:00:23-05:00	debug	source=deployer	pulling module: arcaflow-plugin-wait@git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong
2024-01-09T10:00:26-05:00	debug	source=deployer	pip install stdout: Collecting arcaflow-plugin-wait@ git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong
  Cloning https://github.com/arcalot/arcaflow-plugin-wait.git (to revision afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong) to /private/var/folders/d_/wv7q0mcs6x9bm10d_jx0j8tr0000gn/T/pip-install-y34n34i1/arcaflow-plugin-wait_8bc50fdb2c394bb895a68a5b1d92d562
2024-01-09T10:00:26-05:00	error	source=main	Invalid workflow (invalid workflow (failed to load schema for step long_wait (failed to deploy plugin of deployment type 'python' with source 'arcaflow-plugin-wait@git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong' (error pip installing arcaflow-plugin-wait@git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong (exit status 1)))))
exit status 1
```

However, module name typo is okay:
```
Discarding git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1: Requested arcaflow-plugin-wait from git+https://github.com/arcalot/arcaflow-plugin-wait.git@afdc2323805ffe2b37271f3a852a4ce7ac7379e1 has inconsistent name: expected 'arcaflow-plugin-wait-wrong', but metadata has 'arcaflow-plugin-wait'
```
```
debug	source=deployer	python plugin module stderr: /Users/joconnel/Documents/projects/arcaflow/workflows/test-workflows/python_deployer_wait_simple/tmp/connector_3-11-6_1/arcaflow-plugin-wait-wrong_afdc2323805ffe2b37271f3a852a4ce7ac7379e1/venv/bin/python: No module named arcaflow_plugin_wait_wrong**
```

Reply:
what do you mean by wrong commit hash?

Reply:
I appended `wrong` to the end of the hash.

Reply:
Stdout wasn't being properly output. The errors that are expected are:
```
 WARNING: Did not find branch or tag 'afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong', assuming revision or ref.
```
and
```
 'afdc2323805ffe2b37271f3a852a4ce7ac7379e1wrong' did not match any file(s) known to git
```

This error comes from git, and not our code. So the stderr is all we need.

This is now resolved as of the most recent version.